crates/lox-io/src/ndm/kvn/parser.rs

```diff
@@ -0,0 +1,1013 @@
+/*
+ * Copyright (c) 2023. Helge Eichhorn and the LOX contributors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+// KVN spec section 7.4 of https://public.ccsds.org/Pubs/502x0b3e1.pdf
+
+use nom::bytes::complete as nb;
+use nom::character::complete as nc;
+use nom::error::{ErrorKind, ParseError};
+use nom::sequence as ns;
+use regex::Regex;
+
+use super::deserializer::KvnDeserializerErr;
+
+#[derive(Debug, PartialEq)]
+pub enum KvnStringParserErr<I> {
+    EmptyKeyword { input: I },
+    EmptyValue { input: I },
+    InvalidFormat { input: I },
+    ParserError(I, ErrorKind),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum KvnKeyMatchErr<I> {
```

Comment:
Could this be `pub struct KeywordNotFoundError` rather than a single-variant enum?